resource.go

```diff
@@ -2,194 +2,104 @@ package huma
-	allParams = append(allParams, op.params...)
-	path := r.path
-	for _, p := range allParams {
-		if p.In == "path" {
-			component := "{" + p.Name + "}"
-			if !strings.Contains(path, component) {
-				if !strings.HasSuffix(path, "/") {
-					path += "/"
-				}
-				path += component
-			}
-		}
-	}
-
-	r.router.register(method, path, op)
+// Head creates a new HTTP HEAD operation at this resource.
+func (r *Resource) Head(operationID, docs string, responses ...Response) *Operation {
+	return r.Operation(http.MethodHead, operationID, docs, responses...)
 }
 
-// Head creates an HTTP HEAD operation on the resource.
-func (r *Resource) Head(docs string, handler interface{}) {
-	r.operation(http.MethodHead, docs, handler)
+// Get creates a new HTTP GET operation at this resource.
+func (r *Resource) Get(operationID, docs string, responses ...Response) *Operation {
+	return r.Operation(http.MethodGet, operationID, docs, responses...)
 }
 
-// List is an alias for `Get`.
-func (r *Resource) List(docs string, handler interface{}) {
-	r.Get(docs, handler)
+// Put creates a new HTTP PUT operation at this resource.
+func (r *Resource) Put(operationID, docs string, responses ...Response) *Operation {
+	return r.Operation(http.MethodPut, operationID, docs, responses...)
 }
 
-// Get creates an HTTP GET operation on the resource.
-func (r *Resource) Get(docs string, handler interface{}) {
-	r.operation(http.MethodGet, docs, handler)
+// Patch creates a new HTTP PATCH operation at this resource.
+func (r *Resource) Patch(operationID, docs string, responses ...Response) *Operation {
+	return r.Operation(http.MethodPatch, operationID, docs, responses...)
 }
 
-// Post creates an HTTP POST operation on the resource.
-func (r *Resource) Post(docs string, handler interface{}) {
-	r.operation(http.MethodPost, docs, handler)
+// Delete creates a new HTTP DELETE operation at this resource.
+func (r *Resource) Delete(operationID, docs string, responses ...Response) *Operation {
+	return r.Operation(http.MethodDelete, operationID, docs, responses...)
 }
 
-// Put creates an HTTP PUT operation on the resource.
-func (r *Resource) Put(docs string, handler interface{}) {
-	r.operation(http.MethodPut, docs, handler)
+// Middleware adds a new standard middleware to this resource, so it will
+// apply to requests at the resource's path (including any subresources).
+// Middleware can also be applied at the router level to apply to all requests.
+func (r *Resource) Middleware(middlewares ...func(next http.Handler) http.Handler) {
+	r.mux.Use(middlewares...)
 }
 
-// Patch creates an HTTP PATCH operation on the resource.
-func (r *Resource) Patch(docs string, handler interface{}) {
-	r.operation(http.MethodPatch, docs, handler)
-}
+// SubResource creates a new resource attached to this resource. The passed
+// path will be appended to the resource's existing path. The path can
+// include parameters, e.g. `/things/{thing-id}`. Each resource path must
+// be unique.
```

Comment:
Is there anything enforcing uniqueness, or is this a convention?

Reply:
The underlying router will enforce it and panic when you try to register the same one twice. If this is too much of a burden, we could cache the sub-routers for re-use, but I'd like to tackle that as a future change if we run into it.